ledger/apptxn_test.go

```diff
@@ -58,14 +59,63 @@ func TestPayAction(t *testing.T) {
          itxn_submit
         `))
 
+		// We're going to test some mining effects here too, so that we have an inner transaction example.
+		proposer := basics.Address{0x01, 0x02, 0x03}
+		dl.txns(&txntest.Txn{
+			Type:     "pay",
+			Sender:   addrs[7],
+			Receiver: proposer,
+			Amount:   1_000_000 * 1_000_000, // 1 million algos is surely an eligible amount
+		}, &txntest.Txn{
+			Type:         "keyreg",
+			Sender:       proposer,
+			Fee:          3_000_000,
+			VotePK:       crypto.OneTimeSignatureVerifier{0x01},
+			SelectionPK:  crypto.VRFVerifier{0x02},
+			StateProofPK: merklesignature.Commitment{0x03},
+			VoteFirst:    1, VoteLast: 1000,
+		})
+
 		payout1 := txntest.Txn{
 			Type:          "appl",
 			Sender:        addrs[1],
 			ApplicationID: ai,
 			Accounts:      []basics.Address{addrs[1]}, // pay self
 		}
 
-		dl.fullBlock(&payout1)
+		presink := micros(dl.t, dl.generator, genBalances.FeeSink)
+		preprop := micros(dl.t, dl.generator, proposer)
+		dl.beginBlock()
+		dl.txns(&payout1)
+		vb := dl.endBlock(proposer)
+		// First MiningPct > 0
+		if ver >= 40 {
```

Comment:
NIT - this is something we have to remember - seems unlikely we do a consensus upgrade before this is out though

Reply:
That's intentional. If I computed the first version in which mining is enabled, then the test would not fail if someone changed that someday. I tend to use "magic number" constants in unit tests for things that appear in consensus, in order to avoid someone changing a thing that we aren't allowed to changed, but having the unit tests pass anyway.  They might change it here too, but at least they have to try harder to hurt themself.